### ISSUE TITLE
overflows, negative integers

### ISSUE THREAD

**Issue by skaunov** [noir-lang/docs#262]
### Problem

Searching docs for "overflow" and for "negative" return nothing.

### Happy Case

Would be cool to easily understand overflow behavior and negative numbers from docs.

### Would you like to submit a PR for this Issue?

No

**Comment by critesjosh** [comment]
There isn't native support for signed integers, but you can check if [this](https://github.com/resurgencelabs/signed_int) library will meet your needs.

Also regarding "overflow", are you wondering about the integer type? Integer types are range constrained to the specified size, so an integer that is out of bounds will cause the proof to fail. If that's not clear based on the [docs](https://noir-lang.org/language_concepts/data_types#integers) do you have any suggestions on how to improve?

**Comment by skaunov** [comment]
Yeah, I first of all wanted to feedback the question left (in my case) 
by the docs. Thank you for answering them! I need all of these soak in 
my mind a bit, and if I will be able to propose anything -- will create 
a PR.

**Comment by critesjosh** [comment]
appreciate the feedback!

**Comment by skaunov** [comment]
I hope to propose an update... X)Could you help me to understand why negative numbers are impossible (since they're made as a `struct` in this `lib`), and whether *underflow* halts the provement process as well as overflow?On Jul 21, 2023 11:12 PM, josh crites ***@***.***> wrote:
appreciate the feedback!

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
On Jul 21, 2023 11:12 PM, josh crites ***@***.***> wrote:
appreciate the feedback!

**Comment by critesjosh** [comment]
Negative numbers aren't possible when using the [integer type](https://noir-lang.org/language_concepts/data_types#integers) because the language enforces that they are unsigned.

If you try to generate a proof for a Noir program with values that underflow or overflow, you'll see `Error: could not satisfy all constraints`--you won't even be able to generate the proof.

**Comment by critesjosh** [comment]
I believe we are sticking with libraries for signed integer functionality for now. You can check the details of the noir roadmap [here](https://github.com/orgs/noir-lang/projects/1/views/14).

The BattleZips repo is a good source of inspiration, but yea it's dated at this point.

**Comment by critesjosh** [comment]
I am going to close this issue. Feel free to reference it if you open a PR.
